packages/ibm-products/src/components/InlineTip/InlineTip.js

```diff
@@ -0,0 +1,269 @@
+const blockClass = `${pkg.prefix}--inline-tip`;
+const componentName = 'InlineTip';
+
+// NOTE: the component SCSS is not imported here: it is rolled up separately.
+
+// Default values can be included here and then assigned to the prop params,
+// e.g. prop = defaults.prop,
+// This gathers default values together neatly and ensures non-primitive
+// values are initialized early to avoid react making unnecessary re-renders.
+// Note that default values are not required for props that are 'required',
+// nor for props where the component can apply undefined values reasonably.
+// Default values should be provided when the component needs to make a choice
+// or assumption when a prop is not supplied.
+
+// Default values for props
+const defaults = {
+  closeIconDescription: 'Close',
+  collapsible: false,
+  collapseButtonLabel: 'Read less',
+  expandButtonLabel: 'Read more',
+  narrow: false,
+  onClick: () => {},
+  onClose: () => {},
+};
+
+/**
+ * TODO: A description of the component.
+ */
+export let InlineTip = React.forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+
+      children /* TODO: remove if not needed. */,
+      className,
+      /* TODO: add other props for InlineTip, with default values if needed */
+
+      closeIconDescription = defaults.closeIconDescription,
+      collapsible = defaults.collapsible,
+      collapseButtonLabel = defaults.collapseButtonLabel,
+      expandButtonLabel = defaults.expandButtonLabel,
+      narrow = defaults.narrow,
+      onClick,
+      onClose,
+      tertiaryButtonLabel,
+      contentCTA,
+      title = defaults.title,
+      media,
+
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    const [isCollapsed, setIsCollapsed] = useState(collapsible);
+
+    const previewText = useMemo(
+      () => getComponentText(React.Children.toArray(children)),
+      [children]
+    );
+    let childrenToRender = children;
+
+    if (!media && collapsible && isCollapsed) {
+      childrenToRender = (
+        <p className={`${blockClass}__preview-text`}>{previewText}</p>
+      );
+    }
+
+    // If `collapsible` is changed after initial render...
+    useEffect(() => {
+      setIsCollapsed(collapsible);
+    }, [collapsible]);
+
+    return (
+      <div
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className, // Apply any supplied class names to the main HTML element.
+          [collapsible ? `${blockClass}__collapsible` : null],
+          [isCollapsed ? `${blockClass}__collapsible-collapsed` : null],
+          [narrow ? `${blockClass}__narrow` : `${blockClass}__wide`],
+          [media ? `${blockClass}__has-media` : null]
+        )}
+        ref={ref}
+        role="main"
```

Comment:
`role="main"` is probably too high level here. Probably left this in from the default since I think we saw the same thing when I reviewed `SteppedAnimatedMedia` for Jeff.

Does `InlineTip` make sense as an `aside`? [`role="complementary"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/complementary_role) seems like a decent candidate as an alternative here.